### PR TITLE
Added timeline + tests

### PR DESCRIPTION
**Pull request by albert-nowak**
*JIRA ticket: * [WDP-117](https://kindergarten.stxnext.pl/browse/WDP-117)

out2

out1

### CODE REVIEW

File: src/presence_analyzer/static/presence_start_end.html [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        }
+    </style>
+
+    <script src="/static/js/jquery.min.js"></script>
+    <script type="text/javascript" src="https://www.google.com/jsapi"></script>
+    <script type="text/javascript">
+        google.load("visualization", "1", {packages:["corechart", "timeline"], 'language': 'pl'});
+    </script>
+    <script type="text/javascript">
+        function parseInterval(value) {
```
Comment (dpolcyn) [inline src/presence_analyzer/static/presence_start_end.html]: move it to another file and use in all files

File: src/presence_analyzer/static/presence_start_end.html [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            var result = new Date(value);
+            return result
+        }
+        
+        (function($) {
+            $(document).ready(function(){
```
Comment (dpolcyn) [inline src/presence_analyzer/static/presence_start_end.html]: ) { - space between

File: src/presence_analyzer/static/presence_start_end.html [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                            $.each(result, function(index, value){
+                                value[1] = parseInterval(value[1]);
+                                value[2] = parseInterval(value[2]);
+                            });
+                            var data = new google.visualization.DataTable();
+                            data.addColumn('string', 'Weekday');
+                            data.addColumn({ type: 'datetime', id: 'Start' });
+                            data.addColumn({ type: 'datetime', id: 'End' });
+                            data.addRows(result);
+                            var options = {
```
Comment (dpolcyn) [inline src/presence_analyzer/static/presence_start_end.html:112]: how to declare multiple variables in js
```
var aaa = {
    1: 'a',
    2: 'b',
}, <- coma here
    bbb = 4,
    ccc = 5; <- ; after last item
```

File: src/presence_analyzer/static/presence_start_end.html [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                    <option value="">--</option>
+                </select>
+                <div id="chart_div" style="display: none">
+                </div>
+                <div id="loading">
+                    <img src="/static/img/loading.gif" />
+                </div>
+            </p>
+        </div>
+
```
Comment (dpolcyn) [inline src/presence_analyzer/static/presence_start_end.html]: without empty line

File: src/presence_analyzer/tests.py [hunk 5]
```diff
@@ -2,12 +2,14 @@
 """
 Presence analyzer unit tests.
 """
-import os.path
 import json
+import os.path
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: sort imports

File: src/presence_analyzer/tests.py [hunk 6]
```diff
@@ -53,6 +54,96 @@ def test_api_users(self):
...
+            data,
+            [
+                ['Weekday', 'Presence (s)'],
+                ['Mon', 0],
+                ['Tue', 30047],
+                ['Wed', 24465],
+                ['Thu', 23705],
+                ['Fri', 0],
+                ['Sat', 0],
+                ['Sun', 0]]
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: style

File: src/presence_analyzer/tests.py [hunk 7]
```diff
@@ -53,6 +54,96 @@ def test_api_users(self):
...
+                    'January 1, 2013 10:48:46',
+                    'January 1, 2013 17:23:51'
+                ],
+                [
+                    'Fri',
+                    'January 1, 2013 00:00:00',
+                    'January 1, 2013 00:00:00'
+                ]
+            ]
+
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: without empty line

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -86,6 +177,77 @@ def test_get_data(self):
...
         )
 
+    def test_group_by_weekday(self):
+        """
+        Test grouping dates by weekdays.
+        """
+        data = utils.get_data()
+        weekdays = utils.group_by_weekday(data[10])
+        self.assertEqual(weekdays[1][0], 30047)
+        with self.assertRaises(KeyError):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ? don't test exception

File: src/presence_analyzer/tests.py [hunk 9]
```diff
@@ -86,6 +177,77 @@ def test_get_data(self):
...
+
+    def test_average_hour(self):
+        """
+        Test calculating average hour and correct data string.
+        """
+        data = utils.get_data()
+        weekdays = utils.group_by_weekday(data[10], seconds=True, start=True)
+        self.assertEqual(
+            utils.average_hour(weekdays[1]),
+            "January 1, 2013 09:39:05"
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: '

File: src/presence_analyzer/tests.py (hunk 5 above)
Comment (albert-nowak) [reply src/presence_analyzer/tests.py]: `import datetime`
`import json`
`import os.path`
`import unittest`
Alphabetically, like this? Is it okey, to use auto-sorting sublime plugin?

**Pull request closed without merging**